src/screenshot/screenshot-module.ts

```diff
@@ -72,35 +63,38 @@ export default class ScreenshotModule {
   }
 
   public getScreenshotPath(projectId: number, deploymentId: number) {
-    return path.join(this.getScreenshotDir(projectId, deploymentId), 'screenshot.jpg');
+    return path.join(
+      this.getScreenshotDir(projectId, deploymentId),
+      this.getScreenshotFilename(projectId, deploymentId),
+    );
+  }
+
+  public getScreenshotFilename(_projectId: number, _deploymentId: number) {
+    return 'screenshot.jpg';
   }
 
   public async deploymentHasScreenshot(projectId: number, deploymentId: number) {
-    return await new Promise(resolve => {
-      fs.exists(this.getScreenshotPath(projectId, deploymentId), resolve);
-    });
+    return await new Promise<boolean>(resolve => fs.exists(this.getScreenshotPath(projectId, deploymentId), resolve));
   }
 
-  private getScreenshotterPath(projectId: number, deploymentId: number) {
-    return path.join('/screenshots', String(projectId), String(deploymentId), 'screenshot.jpg');
+  private getRemoteDest(projectId: number, deploymentId: number) {
+    // TODO: the remote path shouldn't be hardcoded here
+    return path.join('/screenshots', String(projectId), String(deploymentId));
   }
 
   /*
    * Take a screenshot for given projectId and deploymentId
    */
   public async takeScreenshot(projectId: number, deploymentId: number, shortId: string) {
     const url = sprintf(this.urlPattern, `${shortId}-${projectId}-${deploymentId}`);
+    const dest = this.getRemoteDest(projectId, deploymentId);
+    const options: PageresOptions = {
+      filename: this.getScreenshotFilename(projectId, deploymentId).replace(/\.[^.]+$/, ''),
```

Comment:
I :heart: regexps, but maybe take out this one into a separate `stripFileExtension` method to make it more explicit what it's doing?